### PR TITLE
[M] CANDLEPIN-923: Optimized content DTO translation

### PR DESCRIPTION
**Pull request by Ceiu**
- Updated the GET /content and GET /owners/.../content endpoints to translate internal model objects to DTOs in bulk to avoid performance issues associated with fetching lazily populated content fields

### CODE REVIEW

**Comment by Ceiu** [comment]
Verifying this PR can (and should) be done in three ways:
- traditional static analysis
- measuring the runtime of the two cited endpoints with a reasonable amount of data
- enabling SQL query logging and verifying fetching content from the two endpoints does not trigger a query against `cp_content_required_products` for each entity returned

SQL query logging can be enabled by setting the following in `candlepin.conf` and restarting Tomcat:
```
log4j.logger.org.hibernate.SQL=DEBUG
```

This won't show parameters, but we aren't interested in those here. We only need to verify that we have a single (or couple) hits against the required products table, rather than the n+1 issue of one per entity.

**Comment by Ceiu** [comment]
As much as it pains me to say this: Sonarcloud is actually right for once. But the more correct approach here is a design and multi-task epic in and of itself, which we simply do not have the time for in this PR and will be addressed in such a future task.

File: src/main/java/org/candlepin/model/ContentCurator.java [hunk 1]
```diff
@@ -717,4 +717,45 @@ WITH RECURSIVE pool_products (product_uuid) AS (
         return output;
     }
 
+    /**
+     * Fetches the required product IDs for the contents specified by the collection of content UUIDs.
+     * The required product IDs will be returned as a mapping of consumer UUID to required product ID
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/model/ContentCurator.java:722]: Update `mapping of consumer UUID` to `mapping of content UUID`

**Review by joshmalbrecht — CHANGES_REQUESTED** [review]
Looks good! I just noticed a small change in one of the Java docs

File: src/main/java/org/candlepin/model/ContentCurator.java (hunk 1 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/model/ContentCurator.java]: You wouldn't believe the number of times I've made this mistake working with content. Fixed.

**Comment by Ceiu** [comment]
Also added missing tests for the content curator addition

**Review by joshmalbrecht — APPROVED** [review]

**Pull request merged**